server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodVersion.java

```diff
@@ -0,0 +1,90 @@
+package org.infinispan.server.hotrod;
+
+import java.util.Arrays;
+
+/**
+ * The various Hot Rod versions
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public enum HotRodVersion {
+   UNKNOWN(0, 0),
+   HOTROD_10(1, 0),
+   HOTROD_11(1, 1),
+   HOTROD_12(1, 2),
+   HOTROD_13(1, 3),
+   HOTROD_20(2, 0),
+   HOTROD_21(2, 1),
+   HOTROD_22(2, 2),
+   HOTROD_23(2, 3),
+   HOTROD_24(2, 4),
+   HOTROD_25(2, 5),
+   HOTROD_26(2, 6),
+   HOTROD_27(2, 7),
+   ;
+
+   private final int major;
+   private final int minor;
+   private final byte version;
+   private final String text;
+
+   HotRodVersion(int major, int minor) {
+      this.major = major;
+      this.minor = minor;
+      this.version = (byte) (major * 10 + minor);
+      this.text = version > 0 ? String.format("HOTROD/%d.%d", major, minor) : "UNKNOWN";
+   }
+
+   public byte getVersion() {
+      return version;
+   }
+
+   public boolean isNewerThan(byte version) {
+      return this.version > version;
+   }
+
+   public boolean isAtLeast(byte version) {
+      return this.version <= version;
+   }
+
+   public String toString() {
+      return text;
+   }
+
+   public static final HotRodVersion LATEST;
+   private static final HotRodVersion[] versions = new HotRodVersion[256];
+
+   static {
+      LATEST = values()[values().length - 1];
+      Arrays.fill(versions, UNKNOWN);
+      for(HotRodVersion version : values()) {
+         versions[version.version] = version;
+      }
+   }
+
+   public static HotRodVersion forVersion(byte version) {
```

Comment:
Also it isn't clear unless I look at the source directly what format this byte should be in when it is invoked. But I guess it is fine as this is just the defined version used by the hotrod protocol.

Reply:
Yeah, the wire format defines the version. I'm just using simple arithmetic (major*10+minor) because that's what has been done up to now, but if we get to 2.10 we're in trouble :)